js/stringProcessing/urlStructure.js

```diff
@@ -0,0 +1,114 @@
+let core = require( "tokenizer2/core" );
+
+var urlTokenizer;
+let tokens;
+
+let staticRegex = /^[^%]+$/;
```

Comment:
I'm missing the code documentation for the regexes. Some explanation what this contains/matches would be nice.